src/main/java/com/zaxxer/sansorm/internal/Introspected.java

```diff
@@ -63,23 +63,23 @@
    private String[] updatableColumns;
 
    /**
-    * Constructor.  Introspect the specified class and cache various annotation
-    * data about it.
+    * Constructor. Introspect the specified class and cache various annotation data about it.
     *
     * @param clazz the class to introspect
     */
    Introspected(Class<?> clazz) {
       this.clazz = clazz;
+      this.columnToField = new TreeMap<>(String.CASE_INSENSITIVE_ORDER); // support both in- and case-sensitive DBs
```

Comment:
this is the main point of the fix. of course using HashMap is faster, but i assume that most tables usually have few columns so that it could hurt performance. another option would be to lowercase all inputs for this map, but that would involve a lot of unnecessary String allocations.